### ISSUE TITLE
Event state should be part of event metadata

### ISSUE THREAD

**Issue by ThorbenLindhauer** [zeebe-io/zeebe#367]
Every event has a property that indicates its current state (e.g. `CREATE`, `CREATED`, etc. for tasks). This property is currently part of the message pack payload. It should be moved into the common SBE header for events, namely `BrokerEventMetadata` (where also things like event type are).

Why? Every event has this and therefore promoting this property into the common header makes building clients easier.

**Comment by ThorbenLindhauer** [comment]
This would also allow us to simply the code that correlates an event to its handler in the broker.

**Comment by berndruecker** [comment]
Pretty important to have it - as currently the method is already there but throws an exception for clients when used (see below).

It would be important to also have constants for the values - as the following constant classes are part of the broker and cannot be used in the java client

``
              //if (WorkflowInstanceState.WORKFLOW_INSTANCE_CREATED.equals(event.getState())) {    
              if ("WORKFLOW_INSTANCE_CREATED".equals(event.getState())) {
...
              //if (DeploymentState.DEPLOYMENT_CREATED.equals(eventType)) {                
              if ("DEPLOYMENT_CREATED".equals(eventType)) {                
``



```
java.lang.RuntimeException: not implemented
	at io.zeebe.client.event.impl.TopicEventImpl.getState(TopicEventImpl.java:69) ~[zeebe-client-java-0.1.0-SNAPSHOT.jar:0.1.0-SNAPSHOT]
	at com.camunda.consulting.zeebe.listener.ZeebeListener.lambda$0(ZeebeListener.java:100) ~[classes/:na]
```

**Comment by ThorbenLindhauer** [comment]
Raft events should then also have a state.

**Comment by menski** [comment]
implemented with #794
